src/Categories/Adjoint/Parametric.agda

```diff
@@ -0,0 +1,150 @@
+module Categories.Adjoint.Parametric where
+
+open import Level
+
+open import Data.Product
+open import Categories.Category.Core using (Category)
+open import Categories.Functor using (Functor; _∘F_) renaming (id to idF)
+open Categories.Functor.Functor using (F₀; F₁; homomorphism; identity; F-resp-≈)
+open import Categories.Adjoint
+open import Categories.Category.Construction.Functors
+open import Categories.Diagram.Cowedge
+open import Categories.NaturalTransformation.Dinatural
+open import Categories.Functor.Bifunctor
+open import Categories.Morphism.Properties
+open import Categories.Functor.Bifunctor.Properties
+open import Categories.NaturalTransformation
+open import Categories.NaturalTransformation.NaturalIsomorphism
+open import Categories.Morphism.Reasoning as MorphismReasoning
+import Categories.Morphism.Reasoning as MR
+open Categories.NaturalTransformation.NaturalTransformation using (η)
+open import Function.Equality using (_⟨$⟩_; _⟶_; cong; _∘_)
+
+open import Categories.Adjoint.Equivalents
+
+private
+  variable
+    o ℓ e : Level
+    C D E : Category o ℓ e
+
+-- A parametric adjunction is a family of functors Lc ⊣ Rc indexed by an object c of
+-- a category C; equivalently, a pair of functors L : C x D → E and R : Cᵒ x E → D
+-- such that for every c in C Lc = L(c,-) ⊣ R(c,-) = Rc.
+record ParametricAdjoint {C D E : Category o ℓ e} (L : Functor C (Functors D E)) (R : Functor (Category.op C) (Functors E D)) : Set (o ⊔ ℓ ⊔ e) where
+  private
+    module C = Category C
+    module D = Category D
+    module E = Category E
+    open MR E
+    module L = Functor L
+    module R = Functor R
+    module R₀ A = Functor (R.F₀ A)
+    module L₀ A = Functor (L.F₀ A)
+  field
+    areAdjoint : ∀ A → Adjoint (L.F₀ A) (R.F₀ A)
+
+  module A c = Adjoint (areAdjoint c)
+
+  {-
+    We seem to need a field ensuring that the adjunction L c ⊣ R c is natural in (c : C),
+    which means that the square
+
+      hom(L(c , x) , y) ----------------→ hom(x , R(c , y))
+        ↑                                          ↑
+        |                                          |
+        |                                          |
+        |                                          |
+      hom(L(c' , x) , y) ---------------→ hom(x , R(c' , y))
+
+    is commutative; unfolding the definitions, it seems that proving this exactly amounts to show
+    that the *unit* η c of the adjunction is a *wedge*. Viceversa, one also needs this naturality
+    when proving that the cowedge condition holds, so they seem to be logically equivalent.
+    Mac Lane (IV.7.Thm 3) seems to claim, however, that this condition needs not to be assumed
+    and can be derived from a better definition of parametric adjunction that we are not
+    currently able to devise.
+
+    More info at: https://math.stackexchange.com/questions/4581092/parametric-adjunctions-induce-cowedges-a-real-proof
+  -}
+
```

Comment:
@JacquesCarette this is the major open point of the PR, otherwise, we believe it is ready for review: do you have any suggestion on how to address this?

Reply:
I was going to say that it's usually better in `agda-categories` to work in terms of unit/counit, but I see that `param-nat` is already phrased that way.

Perhaps there's a lead in https://golem.ph.utexas.edu/category/2012/11/parametrized_mates_and_multiva.html ?